sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -23,29 +23,43 @@ import org.apache.spark.sql.types.{DoubleType, NumericType}
 
 private[sql] object StatFunctions {
   
+  /** Calculate the Pearson Correlation Coefficient for the given columns */
+  private[sql] def pearsonCorrelation(df: DataFrame, cols: Seq[String]): Double = {
+    val counts = collectStatisticalData(df, cols)
+    counts.Ck / math.sqrt(counts.MkX * counts.MkY)
```

Comment:
Oh, they canceled each other. Could you add a non-trivial test to Scala? Now it only has `0.0`, `1.0`, and `-1.0`. In your test, please provide Python/R commands in comments to reproduce the result. This makes easier for others to verify.